src/lib.rs

```diff
@@ -320,7 +320,9 @@ pub async fn initialize_staged(
                         CanonicalFunction::Lower { .. }
                         | CanonicalFunction::ResourceNew { .. }
                         | CanonicalFunction::ResourceDrop { .. }
-                        | CanonicalFunction::ResourceRep { .. } => {
+                        | CanonicalFunction::ResourceRep { .. }
+                        | CanonicalFunction::ThreadSpawn { .. }
+                        | CanonicalFunction::ThreadHwConcurrency => {
```

Comment:
I wasn't sure what the distinction was between a `core_function_count` and a `function_count` to know which match arm this should be in (or if it shouldn't update either)

Reply:
Sounds good!